src/net_processing.cpp

```diff
@@ -1332,8 +1332,11 @@ void PeerLogicValidation::UpdatedBlockTip(const CBlockIndex *pindexNew, const CB
 /**
  * Handle invalid block rejection and consequent peer banning, maintain which
  * peers announce compact blocks.
+ * Called both in case of cursory DoS checks failing (implying the peer is likely
+ * sending us bogus data) and after full validation of the block fails (which may
```

Comment:
a85e8b7

In `ConnectTip`, we call `ConnectBlock` then `BlockChecked` callback independently of success or failure of block validation ?